### PR TITLE
Add iModel extents expansion event / Viewport listener

### LINKED ISSUE #4618: DPTA uses wrong displayedExtents when loading and rendering point cloud tile trees
**Describe the bug**
DPTA uses the wrong displayedExtents for point cloud datasets (most likely could happen to all reality data).

**To Reproduce**

1. Use display-performance-test-app to render a local point cloud data set in PNTS format.
4. Observe that the image snapped by DPTA contains a partially rendered point cloud scene
5. If you put some diagnostic code in place, you can see that the displayedExtents of the iModel get expanded -- but too late in the process -- this is because the displayExtents expands when the tile tree loads and notifies all view-managed viewports (which DPTA does not have).

**Expected behavior**
The point cloud should fully render, using the proper expanded displayedExtents.  Viewports should register an event listener for when display extents expand and react by invalidating the controller.

**Screenshots**
TBD

**Desktop (please complete the applicable information):**

- OS: Tested on MacOS and Windows
- Browser: Electron
- iTwin.js Version: 3.x

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
Fixes #4618

### CODE REVIEW

File: core/frontend/src/IModelConnection.ts [hunk 1]
```diff
@@ -157,6 +157,12 @@ export abstract class IModelConnection extends IModel {
    */
   public readonly onClose = new BeEvent<(_imodel: IModelConnection) => void>();
 
+  /** Event called immediately after *this* IModelConnection has its displayed extents expanded.
+   * @note This event is called only for this IModelConnection.
+   * @beta
```
Comment (markschlosseratbentley) [inline core/frontend/src/IModelConnection.ts:162]: Should this API be marked external (beta), or left internal only?

Comment (pmconne) [reply core/frontend/src/IModelConnection.ts:162]: Internal. The concept of a "displayed extents" for an iModel doesn't make sense. It applies to all spatial views and it never shrinks. I would like to remove it in the future.

File: core/frontend/src/IModelConnection.ts [hunk 2]
```diff
@@ -157,6 +157,12 @@ export abstract class IModelConnection extends IModel {
...
+   */
+  public readonly onDisplayedExtentsExpansion = new BeEvent<(_imodel: IModelConnection) => void>();
```
Comment (pmconne) [inline core/frontend/src/IModelConnection.ts:164]: Remove `_imodel` arg, that's only useful for old-style JS event callbacks.

File: common/api/core-frontend.api.md [hunk 3]
```diff
@@ -4872,6 +4872,8 @@ export abstract class IModelConnection extends IModel {
     // @beta
     readonly onClose: BeEvent<(_imodel: IModelConnection) => void>;
     // @internal
+    readonly onDisplayedExtentsExpansion: BeEvent<(_imodel: IModelConnection) => void>;
```
Comment (pmconne) [inline common/api/core-frontend.api.md:4875]: rerun extract-api

**Review by pmconne — APPROVED** [review]

**Pull request merged**